dvc/remote/base.py

```diff
@@ -719,6 +790,7 @@ def _checkout_dir(
             if self.changed(entry_info, entry_checksum_info):
                 if self.exists(entry_info):
                     self.safe_remove(entry_info, force=force)
+                self.makedirs(entry_info.parent)
```

Comment:
Didn't we discuss that this one is not needed, because `link` will call it anyway?

Reply:
Yes! Good catch, @efiop , it was a leftover